engine/execution/state/delta/view.go

```diff
@@ -119,32 +117,37 @@ func (v *View) RegisterUpdates() ([]flow.RegisterID, []flow.RegisterValue) {
 // This function will return an error if it fails to read from the underlying
 // data source for this view.
 func (v *View) Get(owner, controller, key string) (flow.RegisterValue, error) {
```

Comment:
I find that this method's structure could be streamlined (specifically the repeated code for the spock update):
* overall, we are trying to retrieve a value and we are using `readFunc` as fallback 
* in either case, we are updating the spock secret 


Go stylistic guidelines 1suggests: keep the main logic on the lowest level; put fallback logic in if statements. Following this guideline results is more concise logic, which I _personally_ find is also better to read:

```golang
func (v *View) Get(owner, controller, key string) (flow.RegisterValue, error) {
	var err error
	registerID := flow.NewRegisterID(owner, controller, key)

	value, exists := v.delta.Get(owner, controller, key)
	if !exists {
		value, err = v.readFunc(owner, controller, key)
		if err != nil {
			return nil, fmt.Errorf("get register failed: %w", err)
		}
		v.regTouchSet[registerID.String()] = registerID // capture register touch
		v.readsCount++ // increase reads
	}

	// every time we read a value (order preserving) we update the secret
	// with the registerID only (value is not required)
	err = v.updateSpockSecret(registerID.Bytes())
	if err != nil {
		return nil, fmt.Errorf("adding register read to SPoCK secret failed: %w", err)
	}
	return value, nil
}
```

Reply:
I agree, that's easier to read.